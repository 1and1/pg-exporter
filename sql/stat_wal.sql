select current_setting('server_version_num')::int / 10000 >= 14 as is_pg_14;
show fsync; -- needs to be on for postgresql_wal_sync_count
select pg_stat_reset_shared('wal');

\! curl -sf localhost:9135/metrics | grep "^postgresql_wal_" | sed -e 's/ \([1-9]\|0\.\).*/ NNN/'
