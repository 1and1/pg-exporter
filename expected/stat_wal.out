select current_setting('server_version_num')::int / 10000 >= 14 as is_pg_14;
 is_pg_14 
----------
 t
(1 row)

show fsync; -- needs to be on for postgresql_wal_sync_count
 fsync 
-------
 on
(1 row)

select pg_stat_reset_shared('wal');
 pg_stat_reset_shared 
----------------------
 
(1 row)

\! curl -sf localhost:9135/metrics | grep "^postgresql_wal_" | sed -e 's/ \([1-9]\|0\.\).*/ NNN/'
postgresql_wal_buffers_full_count 0
postgresql_wal_bytes_total NNN
postgresql_wal_fpi_total NNN
postgresql_wal_records_total NNN
postgresql_wal_stats_reset NNN
postgresql_wal_sync_count NNN
postgresql_wal_sync_time_total 0
postgresql_wal_write_count NNN
postgresql_wal_write_time_total 0
